cloud integration: add dbt doc in integration section (#10822)
pingcap/docs: tidb-cloud/dbt.md

---
title: TiDB with dbt
summary: Learn the use cases of dbt in TiDB.
---

# TiDB with dbt

> [Data build tool (dbt)](https://www.getdbt.com/) is a popular open-source data transformation tool that enables analytics engineers to transform data in their warehouses through SQL statements. Through the [dbt-tidb](https://github.com/pingcap/dbt-tidb) plug-in, analytics engineers working with TiDB can directly create forms and match data through SQL without having to think about the process of creating tables or views.

Reviewer: Ask reader to read dbt offical doc first. I think a user can't understand this doc before he read the offical doc

Author: This is a tutorial about dbt-tidb. We have already given the dbt website here, so we need not advice readers to read dbt official doc first.

Here we use the official dbt tutorial as an example to introduce the use of TiDB in dbt together.

The software used in this example and its version requirements:
- TiDB 5.3 or upper
- dbt 1.01 or upper
- dbt-tidb 1.0.0

## Installation

Installing dbt and dbt-tidb requires only one command because dbt is installed as a dependency when we install dbt-tidb.

```shell
pip install dbt-tidb
```

> ## Creating project: jaffle shop

Reviewer: 只是一个想法：咱们的官方文档，要用 jaffle shop 这个例子吗？还是从头开始一个例子？

Author: 从头开始还要准备相关数据、导入数据，反而加重用户尝试的负担。

---
title: TiDB with dbt
summary: Learn the use cases of dbt in TiDB.
---

> # TiDB with dbt

Reviewer 2: Use dbt-tidb to Create Tables and Views for TiDB

[Data build tool (dbt)](https://www.getdbt.com/) is a popular open-source data transformation tool that enables analytics engineers to transform data in their warehouses through SQL statements. Through the [dbt-tidb](https://github.com/pingcap/dbt-tidb) plug-in, analytics engineers working with TiDB can directly create forms and match data through SQL without having to think about the process of creating tables or views.

This document uses the official dbt tutorial as an example to introduce how to use TiDB in dbt together.

The software used in this example and its version requirements:

- TiDB 5.3 or later versions
- dbt 1.01 or later versions
- dbt-tidb 1.0.0

> 

Reviewer 2: Make the following sections steps?

> ## Install dbt

Reviewer 2: ```suggestion
## Step 1: Install dbt
```

|           1 | Michael    | P.        | 2018-01-01  | 2018-02-10        |                2 |                 33.0000 |
|           2 | Shawn      | M.        | 2018-01-11  | 2018-01-11        |                1 |                 23.0000 |
|           3 | Kathleen   | P.        | 2018-01-02  | 2018-03-11        |                3 |                 65.0000 |
|           4 | Jimmy      | C.        | NULL        | NULL              |             NULL |                    NULL |
|           5 | Katherine  | R.        | NULL        | NULL              |             NULL |                    NULL |
|           6 | Sarah      | R.        | 2018-02-19  | 2018-02-19        |                1 |                  8.0000 |
|           7 | Martin     | M.        | 2018-01-14  | 2018-01-14        |                1 |                 26.0000 |
|           8 | Frank      | R.        | 2018-01-29  | 2018-03-12        |                2 |                 45.0000 |
....
```

## Generate the docs

Use the following command to generate a visual document:

```
dbt docs generate
dbt docs serve
```

> The document contains the overall structure of the jaffle_shop project and a description of all the tables and views. You can viewed it at [http://localhost:8080](http://localhost:8080).

Reviewer 2: ```suggestion
The document contains the overall structure of the jaffle_shop project and descriptions of all the tables and views. You can view it at [http://localhost:8080](http://localhost:8080).
```

|        ✅         |       ✅        |      ✅      |      Connection retry       |
|        ✅         |       ✅        |      ✅      |            Grant            |

> **Note:**
>
> * TiDB 4.0 ~ 5.0 does not support [CTE](https://docs.pingcap.com/tidb/dev/sql-statement-with). Avoid using `WITH` in your SQL code.
> * TiDB 4.0 ~ 5.2 does not support creating a [temporary table or view](https://docs.pingcap.com/tidb/v5.2/sql-statement-create-table#:~:text=sec\)-,MySQL%20compatibility,-TiDB%20does%20not).
> * TiDB 4.X does not support using SQL func in `CREATE VIEW`. Avoid using it in your SQL code. You can find more details in [PR 27252](https://github.com/pingcap/tidb/pull/27252).

## Supported functions

Cross-db macros are moved from dbt-utils into dbt-core, so you can use the following functions directly. For more information about how to use them, see [dbt-util](https://github.com/dbt-labs/dbt-utils).

- bool_or
- cast_bool_to_text
- dateadd
- datediff
- date_trunc
- hash
- safe_cast
- split_part
- last_day
- cast_bool_to_text
- concat
- escape_single_quotes
- except
- intersect
- length
- position
- replace
- right

> - listagg (not support yet)

Reviewer 2: ```suggestion
- listagg (not supported yet)
```
